src/restruct_poc/math/quaternion_operations.hpp

```diff
@@ -86,24 +94,28 @@ inline std::array<double, 4> QuaternionCompose(std::array<double, 4> q1, std::ar
     return qn;
 }
 
-template <typename M, typename V>
-KOKKOS_INLINE_FUNCTION void ComputeAxialVector(M m, V v) {
+/// Computes the axial vector of a 3x3 rotation matrix
+template <typename Matrix, typename Vector>
+KOKKOS_INLINE_FUNCTION void ComputeAxialVector(Matrix m, Vector v) {
     v(0) = (m(2, 1) - m(1, 2)) / 2.;
     v(1) = (m(0, 2) - m(2, 0)) / 2.;
     v(2) = (m(1, 0) - m(0, 1)) / 2.;
 }
 
-template <typename V, typename Q>
-KOKKOS_INLINE_FUNCTION void RotationVectorToQuaternion(V phi, Q quaternion) {
+/// Returns a 4-D quaternion from provided 3-D rotation vector, i.e. the exponential map
+template <typename Vector, typename Quaternion>
+KOKKOS_INLINE_FUNCTION void RotationVectorToQuaternion(Vector phi, Quaternion quaternion) {
     const auto angle = Kokkos::sqrt(phi(0) * phi(0) + phi(1) * phi(1) + phi(2) * phi(2));
-    const auto cos_angle = Kokkos::cos(angle / 2.0);
-    const auto factor = (Kokkos::abs(angle) < 1.e-12) ? 0. : Kokkos::sin(angle / 2.0) / angle;
+    const auto cos_angle = Kokkos::cos(angle / 2.);
+    const auto factor = (Kokkos::abs(angle) < 1.e-12) ? 0. : Kokkos::sin(angle / 2.) / angle;
+
     quaternion(0) = cos_angle;
-    for (int i = 0; i < 3; ++i) {
-        quaternion(i + 1) = phi(i) * factor;
-    }
+    quaternion(1) = phi(0) * factor;
+    quaternion(2) = phi(1) * factor;
+    quaternion(3) = phi(2) * factor;
```

Comment:
Leave this as a loop.  It gives the compiler some hope for vectorizing the loop and communicates that those components are handled the same way.